modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -398,16 +397,21 @@ private void deletePartition(OperationPurpose purpose, List<String> partition, D
             } catch (MultiObjectDeleteException e) {
                 // We are sending quiet mode requests so we can't use the deleted keys entry on the exception and instead
                 // first remove all keys that were sent in the request and then add back those that ran into an exception.
-                logger.warn(
-                    () -> format(
-                        "Failed to delete some blobs %s",
-                        e.getErrors()
+                logger.warn(() -> {
```

Comment:
I guess it makes sense to not spit out all of the blob names. But two nits: could we instead just pick the first N items rather than truncating it like this? Also since this is a WARN for an error case, having the runnable for the logger isn't really helping much, just makes it harder to read, IMO. Can we pull that out?

Reply:
Pushe 5d67feb41bc629711c1b47ae60caacae310f439d which should address both suggestions. Thanks!

Reply:
Btw the usage of `Strings.collectionToDelimitedStringWithLimit` was modelled after [this](https://github.com/elastic/elasticsearch/blob/6646af015a49c9a47404346a7fd6cdbffd285d5f/server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java#L2511-L2519). It might be worthwhile to extract a new common method for both as well.